contracts/ColonyNetworkStaking.sol

```diff
@@ -69,24 +69,29 @@ contract ColonyNetworkStaking is ColonyNetworkStorage, DSMath {
   {
     reputationRootHash = newHash;
     reputationRootHashNNodes = newNNodes;
-    // Clear out the inactive reputation log. We're setting a new root hash, so we're done with it.
-    delete reputationUpdateLogs[(activeReputationUpdateLog + 1) % 2];
-    // The active reputation update log is now switched to be the one we've just cleared out.
-    // The old activeReputationUpdateLog will be used for the next reputation mining cycle
-    activeReputationUpdateLog = (activeReputationUpdateLog + 1) % 2;
     // Reward stakers
-    rewardStakers(stakers);
-    reputationMiningCycle = 0x0;
+    currentReputationMiningCycle = 0x0;
     startNextCycle();
+    rewardStakers(stakers);
   }
 
   function startNextCycle() public {
-    require(reputationMiningCycle == 0x0);
-    reputationMiningCycle = new ReputationMiningCycle();
+    require(currentReputationMiningCycle == 0x0);
+    currentReputationMiningCycle = nextReputationMiningCycle;
+    if (currentReputationMiningCycle == 0x0) {
```

Comment:
But we `require` that to be `0x0` two lines up?

Reply:
And we set it to `nextReputationMiningCycle` on the line in between, which will be `0x0` the very first time this function is run to kick off reputation mining, and never again.